### ISSUE TITLE
Can't encode a record containing a union annotated with @AvroNamespace

### ISSUE THREAD

**Issue by bobbyrauchenberg** [sksamuel/avro4s#350]
Hi

I've found an issue where an exception is thrown if we try to encode a record with a union annotated with `@AvroNamespace`

The following test will currently fail with an exception

```
test("support encoding a union with @AvroNamespace at the field level") {
    val schema = AvroSchema[Cupcat]
    val cupcat = Cupcat(Option(Snoutley("snoutley")))

    val record = ToRecord[Cupcat](schema).to(cupcat).asInstanceOf[ImmutableRecord]

    val snoutley: Schema = SchemaBuilder.record("Snoutley").namespace("cup.cat").fields().requiredString("name").endRecord()


    record.getSchema shouldBe SchemaBuilder.record("Cupcat").namespace("com.sksamuel.avro4s.record.encoder")
        .fields()
        .name("snoutley").`type`(SchemaBuilder.unionOf().nullType().and().`type`(snoutley).endUnion()).noDefault()
        .endRecord()

    record.values.size shouldBe 1
  }
```

for the record
```
case class Snoutley(name: String)
case class Cupcat(@AvroNamespace("cup.cat") snoutley: Option[Snoutley])
```

I've got a fix for it, and can raise a PR - i think the issue is inside `Encoder.scala`, where the schemas are combined - currently this is done using a `forEach` which doesn't actually combine the schemas. 

My suggested fix, which makes the test pass, just folds over the schemas.

I'll raise a PR and see what you think?

**Comment by sksamuel** [comment]
Good fix.
You can test it with the automatically deployed snapshots.

**Comment by bobbyrauchenberg** [comment]
Looks good, everything in our codebase works against this change

**Comment by bobbyrauchenberg** [comment]
@sksamuel will that snapshot tag always point at the same artifact? 

i.e. can we release code against it?

**Comment by sksamuel** [comment]
Yes it won't change.

**Comment by bobbyrauchenberg** [comment]
Great, thanks!

**Comment by sksamuel** [comment]
Released 3.0.0 final which contains your fix for this.
